src/merkle_tree.mli

```diff
@@ -110,13 +120,13 @@ module Checked
 
   (* TODO: Change [prev] to be [prev_hash : Hash.var] since there may be no need
     to certify that the hash of the element is a particular value. *)
-
   val modify_req :
-       depth:int
+       is_chain_voting:Boolean.var
+    -> depth:int
     -> Hash.var
     -> Address.var
     -> f:(Elt.var -> (Elt.var, 's) Checked.t)
-    -> (Hash.var, 's) Checked.t
+    -> (Hash.var * Elt.var, 's) Checked.t
```

Comment:
I'm not sure that overwriting this is the way we should go here, a new name (`fetch_and_update`?) and keeping an implementation of the original would cause less friction. Would be good to have a doc-comment clarifying the return value too.